app/services/FinancialDataService.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import connectors.FinancialDataConnector
+import models.financialdata.{Charge, FinancialDataQueryParameters, FinancialDataResponse}
+import models.Period
+import uk.gov.hmrc.domain.Vrn
+
+import java.time.LocalDate
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class FinancialDataService @Inject()(
+                                      financialDataConnector: FinancialDataConnector)(implicit ec: ExecutionContext) {
+
+  def getFinancialData(vrn: Vrn, commencementDate: LocalDate): Future[Option[FinancialDataResponse]] =
+    financialDataConnector.getFinancialData(vrn, FinancialDataQueryParameters(fromDate = Some(commencementDate), toDate = Some(LocalDate.now())))
+
+  def getCharge(vrn: Vrn, period: Period): Future[Option[Charge]] = {
+    getFinancialData(vrn, period.firstDay).map { maybeFinancialDataResponse => // TODO check period.first day makes sense, what if someone overpaid a previous period, how would that be represented?
```

Comment:
Yep I think it should be fine to use first day of period. I am almost thinking that maybe we could use last day of period aswell. Its an outstanding question for Tim on how them dates work and if theyre inclusive. But the single charge would be created for the period regardless of the paymount amount on any other periods. So we can delete the TODO